offchain/advance-runner/src/error.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright Cartesi Pte. Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use snafu::Snafu;
+
+use crate::{broker, config, runner, server_manager};
+
+use crate::snapshot::disabled::SnapshotDisabledError;
+use crate::snapshot::fs_manager::FSSnapshotError;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)))]
+pub enum AdvanceRunnerError {
```

Comment:
I feel like this should hoist the error upwards (like RunnerError did):

`pub enum AdvanceRunnerError<SnapError: snafu::Error + 'static>`